test/ecto/embedded_test.exs

```diff
@@ -63,6 +79,12 @@ defmodule Ecto.EmbeddedTest do
     assert %UUIDSchema{uuid: ^uuid, authors: [%Author{}]} =
              Ecto.embedded_load(UUIDSchema, %{"uuid" => uuid, "authors" => [%{}]}, :json)
 
+    assert %Settings{dark_mode: false, default_post: %Post{}} =
+             Ecto.embedded_load(Settings, %{}, :json)
```

Comment:
What happens if I just do `%Settings{}`? Is the default still `nil`? Do we want to force it to be %Post{} even then?

Reply:
It behaves exactly like `:default` on a field. So it will set to `%Post{}`.

Reply:
```elixir
assert %Settings{dark_mode: false, default_post: %Post{}} = %Settings{}
```

Reply:
Let's add a test for this scenario and I believe we are good to go!

Reply:
I was about to approve but I think we're still missing this test 

```elixir
assert %Settings{dark_mode: false, default_post: %Post{}} = %Settings{}
```

Reply:
This is being tested through the load_embedded function, which relies on the struct default. I did it this way because it was how it was already been done. Of course the load is broader and also involves calling the field's load function, but I think that with both the case for non defined and explicit nil we cover all the cases. However, if you think we still need that test I can add it no problem.